rubix-prestosql/src/main/java/com/qubole/rubix/prestosql/PrestoClusterManager.java

```diff
@@ -12,300 +12,51 @@
  */
 package com.qubole.rubix.prestosql;
 
-import com.google.common.base.Throwables;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Lists;
-import com.google.common.reflect.TypeToken;
-import com.google.gson.Gson;
-import com.qubole.rubix.common.utils.ClusterUtil;
-import com.qubole.rubix.spi.ClusterManager;
+import com.qubole.rubix.spi.AsyncClusterManager;
 import com.qubole.rubix.spi.ClusterType;
 import io.prestosql.spi.Node;
 import io.prestosql.spi.NodeManager;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.conf.Configuration;
 
-import javax.annotation.Nullable;
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.lang.reflect.Type;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
-import java.net.URI;
-import java.net.URL;
 import java.net.UnknownHostException;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Objects;
 import java.util.Set;
-import java.util.stream.Collectors;
-
-import static java.util.Objects.requireNonNull;
 
 /**
  * Created by stagra on 14/1/16.
  */
-public class PrestoClusterManager extends ClusterManager
+public class PrestoClusterManager extends AsyncClusterManager
 {
-  private static final String DEFAULT_USER = "rubix";
-  private int serverPort = 8081;
-  private String serverAddress = "localhost";
-
-  private Log log = LogFactory.getLog(PrestoClusterManager.class);
-
-  @Nullable
-  private static volatile NodeManager nodeManager;
-  public static String serverPortConf = "caching.fs.presto-server-port";
-
-  // Safe to use single instance of HttpClient since Supplier.get() provides synchronization
-  @Override
-  public void initialize(Configuration conf)
-          throws UnknownHostException
-  {
-    super.initialize(conf);
-    this.serverPort = conf.getInt(serverPortConf, serverPort);
-    this.serverAddress = ClusterUtil.getMasterHostname(conf);
-  }
-
-  @Override
-  public List<String> getNodesInternal()
-  {
-    if (nodeManager != null) {
-      return getNodesFromNodeManager();
-    }
-
-    try {
-      URL allNodesRequest = getNodeUrl();
-      URL failedNodesRequest = getFailedNodeUrl();
```

Comment:
This part needs to be retained. For the mode of rubix running as separate process NodeManager object will not be available from Presto.

Reply:
Missed StandaloneNodeManager part.